File: lab3/zadanie1/python/packages/primenumbers.py
def compute_sieve(sieve: list, n: int):
	for i in range(n+1):
		if sieve[i]:
			j = i * i
			while j < len(sieve):
				sieve[j] = False
				j = j + i

def valid_number(n):
	if n.isdigit():
		n = int(n)
		if n > 1:
			return n 
		elif n in [0,1]:
			print(f"Nie istnieją żadne liczby pierwsze niewiększe od {n}")
			return False
	print("Drugi argument powinien być liczbą naturalną")
	return False

File: lab3/zadanie1/c/primenumbers.c
#include "primenumbers.h"
#include <ctype.h>
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

void compute_sieve(bool s[], unsigned n){
  unsigned i , j;
  for (i=2; i <= n; i++) s[i] = true;
  for (i=2; i<=n; i++) {
    if (s[i]){
      for (j=i*i; j <= n; j+=i){
        s[j] = false;
      }
    }
  }
}

unsigned count_primes(bool s[], unsigned n){
  unsigned i, c = 0;
  for (i=2; i <= n; i++){
    if (s[i]) c++;
  }
  return c;
} 

bool valid_number(char n[]){
  int len = strlen(n);
  for (int i=0; i < len; i++) {
    if (!isdigit(n[i])){
      printf("Argument powinien być liczbą naturalną\n");
      return false;
    }
  }
  return true;
}
